packages/x-ui-collections-bundle/src/graphql/Collection.ts

```diff
@@ -749,4 +749,31 @@ export abstract class Collection<T = null> {
   protected isUpdateModifier(element: any): element is UpdateFilter<T> {
     return element["$set"];
   }
+
+  /**
+   * Integration with native `useQuery` from Apollo, offering a hybrid approach so apollo re-uses your cache.
+   */
+  public optimistic(_id: ObjectId | string, document: T) {
+    const stringId = _id instanceof ObjectId ? _id.toString() : _id;
+    const operationName = ""; // ! how do we determine this here ?
+
+    // We can't be optimistic if we do no know the operationName
+    if (!operationName) return undefined;
+
+    const collectionName = this.getName?.();
+
+    // We can't be optimistic if we do no know the __typename
+    if (!collectionName) return undefined;
+
+    return {
+      [operationName]: {
+        _id: stringId,
+        __typename:
+          collectionName[collectionName.length - 1] === "s"
```

Comment:
why?

Reply:
With what I said above regarding `__typename`, we can infer it from the collection name, just need to remove the extra `s`